staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go

```diff
@@ -378,3 +385,186 @@ func (o *ExposeServiceOptions) RunExpose(cmd *cobra.Command, args []string) erro
 	}
 	return nil
 }
+
+func (o *ExposeServiceOptions) createService() (*corev1.Service, error) {
+	if len(o.Selector) == 0 {
+		return nil, fmt.Errorf("selector must be specified")
+	}
+	selector, err := parseLabels(o.Selector)
+	if err != nil {
+		return nil, err
+	}
+
+	var labels map[string]string
+	if len(o.Labels) > 0 {
+		labels, err = parseLabels(o.Labels)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	name := o.Name
+	if len(name) == 0 {
+		name = o.DefaultName
+		if len(name) == 0 {
+			return nil, fmt.Errorf("name must be specified")
+		}
+	}
+
+	var portProtocolMap map[string]string
+	if o.Protocols != "" {
+		portProtocolMap, err = parseProtocols(o.Protocols)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// ports takes precedence over port since it will be
+	// specified only when the user hasn't specified a port
+	// via --port and the exposed object has multiple ports.
+	var portString string
+	portString = o.Ports
+	if len(o.Ports) == 0 {
+		portString = o.Port
+	}
+
+	ports := []corev1.ServicePort{}
+	if len(portString) != 0 {
+		portStringSlice := strings.Split(portString, ",")
+		servicePortName := o.PortName
+		for i, stillPortString := range portStringSlice {
```

Comment:
Probably parts of this validation could be in `Validate` but I'll leave it up to you to figure this one out.